Beitetsu Parcel, items_usable and .lua to function

apatrickm opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
Adds the Beitetsu Parcel to items_usable, and adds beitetsu_parcel.lua to add Beitetsu same as retail
<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

## Steps to test these changes
Use a beitetsu parcel
<!-- Clear and detailed steps to test your changes here -->

apatrickm committed:
Add files via upload

apatrickm committed:
Update item_usable.sql

Make the beitetsu parcel usable

WinterSolstice8 commented on sql/item_usable.sql:
```diff
@@ -1857,6 +1857,7 @@ INSERT INTO `item_usable` VALUES (6176,'copy_of_astral_homeland',1,1,117,0,0,0,0
 INSERT INTO `item_usable` VALUES (6177,'copy_of_life-form_study',1,1,117,0,0,0,0,0);
 INSERT INTO `item_usable` VALUES (6178,'copy_of_hrohjs_record',1,1,117,0,0,0,0,0);
 INSERT INTO `item_usable` VALUES (6179,'copy_of_the_bell_tolls',1,1,117,0,0,0,0,0);
+INSERT INTO `item_usable` VALUES (6181,'Beitetsu_Parcel',1,1,24,0,0,0,0,0);
```
lowercase please

WinterSolstice8 commented on scripts/globals/items/beitetsu_parcel.lua:
```diff
@@ -0,0 +1,22 @@
+    target:addItem(4060, math.random(3, 15))
+end
+
+return item_object
```
no newline at end of file

apatrickm committed:
Update beitetsu_parcel.lua

apatrickm committed:
corrected casing

changed beitetsu_parcel to all lower case

apatrickm committed:
Update beitetsu_parcel.lua

zach2good commented on scripts/globals/items/beitetsu_parcel.lua:
```diff
@@ -0,0 +1,23 @@
+end
+
+item_object.onItemUse = function(target)
+    target:addItem(4060, math.random(3, 15))
```
Please don't use a magic number here, add an entry in `items.lua`, add the relevant require to the top of this file, and replace the magic number with `xi.items.BEITETSU`

zach2good commented on scripts/globals/items/beitetsu_parcel.lua:
```diff
@@ -0,0 +1,23 @@
+end
+
+return item_object
+
```
From CI:
```
scripts/globals/items/beitetsu_parcel.lua:23: Found multiple newline characters next to each other, please replace these with single newlines.
```

zach2good replied:
You had it correct in this commit: https://github.com/LandSandBoat/server/pull/2677/commits/f29ac19fcaebbde2066f3cf7f5ab262e15fe0c0c

apatrickm committed:
Update beitetsu_parcel.lua

apatrickm committed:
Update beitetsu_parcel.lua

apatrickm committed:
Update beitetsu_parcel.lua

Changed item ID referance to xi.items.BEITETSU

zach2good commented on scripts/globals/items/beitetsu_parcel.lua:
```diff
@@ -0,0 +1,22 @@
+-- Beitetsu Parcel
+-- Breaks up a Beitetsu Parcel
+-----------------------------------
+require("scripts/globals/msg")
+-----------------------------------
```
```suggestion
require('scripts/globals/items')
require("scripts/globals/msg")
-----------------------------------
```

zach2good commented on scripts/globals/items/beitetsu_parcel.lua:
```diff
@@ -0,0 +1,22 @@
+end
+
+item_object.onItemUse = function(target)
+    target:addItem(xi.items.BEITETSU, math.random(3, 15))
```
`xi.items.BEITETSU` doesn't exist in `scripts/globals/items.lua`, you need to add this line to it:
```lua
    BEITETSU                        = 4060,
```

zach2good replied:
While you're there, you can also add an entry for `BEITETSU_PARCEL` (id 6181)

apatrickm committed:
Update scripts/globals/items/beitetsu_parcel.lua

Co-authored-by: Zach Toogood <zrtoogood@gmail.com>

apatrickm committed:
Update items.lua

Added BEITETSU_PARCEL                 = 6181, and  BEITETSU                        = 4060,

apatrickm committed:
Merge branch 'LandSandBoat:base' into base

apatrickm committed:
Arise and Inundation

Adding Scrolls of Arise and Inundation, along with lua for inundation, arise already has lua

WinterSolstice8 commented on scroll_of_arise.lua:
```diff
@@ -0,0 +1,16 @@
+local item_object = {}
+
+item_object.onItemCheck = function(target)
+    return target:canLearnSpell(494)
```
494 -> xi.magic.spell.ARISE

WinterSolstice8 commented on scroll_of_inundation.lua:
```diff
@@ -0,0 +1,16 @@
+local item_object = {}
+
+item_object.onItemCheck = function(target)
+    return target:canLearnSpell(879)
```
879 -> xi.magic.spell.INUNDATION

apatrickm committed:
Update scroll_of_arise.lua

Corrected magic number

apatrickm committed:
Update scroll_of_arise.lua

apatrickm committed:
Update scroll_of_inundation.lua

corrected magic number

WinterSolstice8 reviewed (approved):
LGTM though needs merge conflict fixed

apatrickm committed:
Merge branch 'base' into base

zach2good commented:
These are not relevant to this PR and should be removed:
- scroll_of_arise.lua
- scroll_of_inundation.lua

apatrickm commented:
The other 2 scroll lua's were added because I can only put in 1 request at a time

zach2good commented:
Replacing this PR with https://github.com/LandSandBoat/server/pull/2819, https://github.com/LandSandBoat/server/pull/2821, and https://github.com/LandSandBoat/server/pull/2820

zach2good commented:
In future, please look up how to use git and GitHub from your desktop, not the web client, and how to squash your commits together (or rebase, if you're feeling brave)
